src/Appium.Net/Appium/Android/UiAutomator/AndroidUiScrollable.cs

```diff
@@ -0,0 +1,402 @@
+        /// <param name="maxSwipes"></param>
+        /// <remarks>https://developer.android.com/reference/android/support/test/uiautomator/UiScrollable#flingtobeginning</remarks>
+        public TerminatedStatementBuilder FlingToBeginning(int maxSwipes)
+        {
+            maxSwipes.RequireIsPositive(nameof(maxSwipes));
+            var forkedBuilder = new StringBuilder(_builder.ToString());
+            forkedBuilder.AppendFormat(".flingToBeginning({0})", maxSwipes);
+            return new TerminatedStatementBuilder(forkedBuilder);
+        }
+
+        /// <summary>
+        /// Performs a fling gesture to reach the end of a scrollable layout element. The end can be at the
+        /// bottom-most edge in the case of vertical controls, or the right-most edge for horizontal controls.
+        /// Make sure to take into account devices configured with right-to-left languages like Arabic and Hebrew.
+        /// Maps to the UiScrollable.flingToEnd() method.
+        /// </summary>
+        /// <param name="maxSwipes"></param>
+        /// <remarks>https://developer.android.com/reference/android/support/test/uiautomator/UiScrollable#flingtoend</remarks>
+        public TerminatedStatementBuilder FlingToEnd(int maxSwipes)
+        {
+            maxSwipes.RequireIsPositive(nameof(maxSwipes));
+            var forkedBuilder = new StringBuilder(_builder.ToString());
+            forkedBuilder.AppendFormat(".flingToEnd({0})", maxSwipes);
+            return new TerminatedStatementBuilder(forkedBuilder);
+        }
+
+        /// <summary>
+        /// Searches for a child element in the present scrollable container. The search first looks for a
+        /// child element that matches the selector you provided, then looks for the content-description in
+        /// its children elements. If both search conditions are fulfilled, the method returns a {@ link UiObject}
+        /// representing the element matching the selector (not the child element in its subhierarchy containing
+        /// the content-description). By default, this method performs a scroll search.
+        /// Maps to the UiScrollable.getChildByDescription(UiSelector, String, boolean) method.
+        /// </summary>
+        /// <param name="uiSelector">
+        /// UiSelector for a child in a scollable layout element
+        /// </param>
+        /// <param name="description">
+        /// Content-description to find in the children of the childPattern match (may be a partial match)
+        /// </param>
+        /// <param name="allowScrollSearch">
+        /// Set to true if scrolling is allowed
+        /// </param>
+        /// <remarks>https://developer.android.com/reference/android/support/test/uiautomator/UiScrollable#getchildbydescription_1</remarks>
+        public TerminatedStatementBuilder GetChildByDescription(AndroidUiSelector uiSelector, string description, bool allowScrollSearch = true)
+        {
+            uiSelector.RequireNotNull(nameof(uiSelector));
```

Comment:
This is a good idea to make it an extension method. I like it. Btw, do we need this argument at all. Isn't it possible to to do something like `nameof(this)` inside the extension method?

Also, like I said, we could save one line of code by calling the verification directly when the corresponding var is used:

```
var forkedBuilder = new StringBuilder(_builder.ToString());
             forkedBuilder.AppendFormat(".getChildByDescription({0}, \"{1}\", {2})", 
                 uiSelector.RequireNotNull().Build(), 
                 description.RequireNotNull(),
                 allowScrollSearch.ToString().ToLowerInvariant());
             return new TerminatedStatementBuilder(forkedBuilder);

```

Reply:
one more concern: `uiSelector.RequireNotNull(nameof(uiSelector));` is going to throw NPE without calling `RequireNotNull` if `uiSelector` is null. That is why exactly this check cannot be an extension, but a separate function

Reply:
> Btw, do we need this argument at all. Isn't it possible to to do something like nameof(this) inside the extension method?

Unfortunately, it is required. There is no native way to automatically pass along the *real* name of the argument. Putting the `nameof()` in the extension method would always report the same variable name (in our case, "value"). This leads to quite a bit of confusion when a single method has several nullable arguments. Example in code:

```csharp
public static T RequireNotNull<T>(T value)
{
    if (value != null) return;
    throw new ArgumentNullException(nameof(value));
}

// The if either argument is null, the exception will report
// that the variable named "value" is null, leading to some confusion
// over which arg was actually null
public void aaaaa(AndroidUiSelector selector, string name)
{
    selector.RequireNotNull();
    name.RequireNotNull();
}
```

> Also, like I said, we could save one line of code by calling the verification directly when the corresponding var is used:

I decided to keep the guard clauses up front since otherwise we will always do the work of cloning the string builder, regardless of whether we *should* do the work. If a guard clause fails, we have to abandon that cloning work, wasting cycles. Admittedly, it usually won't be that much work that gets wasted (fairly small strings) and ideally wouldn't happen often. With that said, if you would still like those cases in-lined, let me know and I'll make the change.

> one more concern: uiSelector.RequireNotNull(nameof(uiSelector)); is going to throw NPE without calling RequireNotNull if uiSelector is null. That is why exactly this check cannot be an extension, but a separate function

In C#, extension functions compile to static method calls. It will still be safe to call RequireNotNull on a null value.

An example call, shown in CIL:
`IL_0007: call string OpenQA.Selenium.Appium.GuardClauses::RequireNotNull<string>(!!0,  string)`

Reply:
> If a guard clause fails, we have to abandon that cloning work, wasting cycles. 

I wouldn't worry about that. Let the interpreter to do micro-optimizations. 

> In C#, extension functions compile to static method calls. It will still be safe to call RequireNotNull on a null value.

Oh, magic, magic everywhere.

Reply:
In-lined final set of guard clauses.